engine/src/mbliphoneplayer.mm

```diff
@@ -718,15 +706,36 @@ - (id)initWithInstance:(MCNativePlayerControl*)instance
 	return self;
 }
 
-- (void)dealloc
+// PM-2015-02-12: [[ Bug 14525 ]] Moved the overlay code entirely into the delegate
+- (void)beginWithOverlay:(bool)p_overlay
+{
+    if (p_overlay)
+    {
+        // The movie's window is the one that is active
+        UIWindow *t_window = [[UIApplication sharedApplication] keyWindow];
+
+        // Now we create an invisible control with the same size as the window
+        m_overlay = [[UIControl alloc] initWithFrame: [t_window frame]];
+        
+        // We want to get notified whenever the overlay control is touched
+        [m_overlay addTarget: self action: @selector(playerWindowTouched:) forControlEvents: UIControlEventTouchDown];
+        [t_window addSubview: m_overlay];
+    }
+}
+
+- (void)end
 {
     if (m_overlay != nil)
     {
         [m_overlay removeTarget: self action: @selector(playerWindowTouched:) forControlEvents: UIControlEventTouchDown];
         [m_overlay removeFromSuperview];
         [m_overlay release];
+        m_overlay = nil;
     }
-    
+}
+
+- (void)dealloc
```

Comment:
I think you need to call 'end' in here - it is entirely plausible that the object might get released within playerPlaybackDidFinish occurring.